mypy/subtypes.py

```diff
@@ -295,8 +295,9 @@ def visit_instance(self, left: Instance) -> bool:
             if call:
                 return self._is_subtype(call, right)
             return False
-        else:
-            return False
+        if isinstance(right, TypeVarType) and right.values:
+            return any(self._is_subtype(left, r_val) for r_val in right.values)
```

Comment:
This is not the correct fix. Now this invalid code generates no error:
```py
from typing import AnyStr, Generic

class Foo(Generic[AnyStr]):
    def method1(self, s: AnyStr, t: AnyStr) -> None:
        print(s)

class Bar(Foo[AnyStr]):
    def method1(self, s: AnyStr, t: AnyStr) -> None:
        print('Child before')
        super().method1('x', b'y')  # Should be an error
        print('Child after')
``` 

We type check the method by substituting all instances of `AnyStr` in the method first with `str`, and later with `bytes`. Currently we don't perform the substitution in `SuperExpr`, which seems to be the root cause of the problem. 

`SuperExpr` should perhaps be modified to support type variable substitution somehow. For example, we could add a list of (type variable, substitution) tuples as an attribute of `SuperExpr` that are initially empty, but as substitutions are performed, we'd add an item to the list. When inferring the type of `SuperExpr`, we'd perform these substitutions. This way the type of `super()` could effectively be treated as `Foo[str]` or `Foo[bytes]` instead of `Foo[AnyStr]`, which should make the original error go away.

Reply:
Ok, I see 😞 

I will refactor this one to expand `SuperExpr()` 👍

Reply:
One more idea: we can represent `method1` as an `@overload` of two functions:
- `def method1(self, s: str, t: str) -> None:`
- `def method1(self, s: bytes, t: bytes) -> None:`

So, this will help us to represent correct argument combinations.
What do you think?